fix order of generic types in MergeWithKey

Jabher opened the pull request:
Flow (at least 0.82+) is producing the following errors on current typing, unless this fix is applied:

```

Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1431:23

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1431|     A: { [k: string]: T },
                               ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1432:23

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1432|     B: { [k: string]: T },
                               ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1439:36

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1439|     (<S: string, A: { [k: string]: T }, B: { [k: string]: T }, T>(
                                            ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1439:59

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1439|     (<S: string, A: { [k: string]: T }, B: { [k: string]: T }, T>(
                                                                   ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1442:36

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1442|     (<S: string, A: { [k: string]: T }, B: { [k: string]: T }, T>(
                                            ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1442:59

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1442|     (<S: string, A: { [k: string]: T }, B: { [k: string]: T }, T>(
                                                                   ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1445:36

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1445|     (<S: string, A: { [k: string]: T }, B: { [k: string]: T }, T>(
                                            ^


Error --------------------------------------------------------------------------- flow-typed/npm/ramda_v0.x.x.js:1445:59

Cannot use function type as a type because function type is a value. To get the type of a value use `typeof`.

   1445|     (<S: string, A: { [k: string]: T }, B: { [k: string]: T }, T>(
                                                                  
```

villesau commented:
@Jabher looks like the tests are failing. This is also conflicting with master. Is this still relevant?

Jabher commented:
Yeah, totally.

It is not issue on me - it is issue on the whole bundle. Errors emerge only on flow 0.82+, and this definition comes with earlier versions.

I've rebased it, transfered changes to newer version, added bits of fixes, but it is still not working anyway. I'll try to think out how can it be fixed, it will take some time from me to fix this.

AdamWills commented:
FYI - a PR fixing broken Ramda tests was just merged: https://github.com/flow-typed/flow-typed/pull/3021

AndrewSouthpaw commented:
This now has merge conflicts that'll need to be resolved.

AndrewSouthpaw commented:
Closing due to inactivity. Feel free to respond to reopen.
